### PR TITLE
fix for filterControl clear function

### PR DESCRIPTION
**Pull request by ioctaptceb**
related issue: #1747 

Solution: use the timeout for a clear cookie function, that searches for and removes all bs.table cookies for a given table. more detail in the ticket

### CODE REVIEW

**Comment by djhvscf** [comment]
Can you provide a fiddle with this change?? I mean in order to test it and be sure that this change works perfectly @ioctaptceb

**Comment by ioctaptceb** [comment]
http://jsfiddle.net/ov75xrjc/1/

to test: type '3' in the star column. list will filter
reload the page, list should still be filtered
click the garbage, should not be filtered
reload page, list should not be filtered

**Comment by ioctaptceb** [comment]
also here is a fiddle for the bug:
http://jsfiddle.net/g4jxgc33/2/

to test: type '3' in the star column. list will filter
reload the page, list should still be filtered
click the garbage, should not be filtered
reload page, list should not be filtered but is **still filtered**

**Comment by ioctaptceb** [comment]
this also allows clearing sort order to a default

**Comment by djhvscf** [comment]
Great!! @ioctaptceb  thanks!

File: src/extensions/filter-control/bootstrap-table-filter-control.js [hunk 1]
```diff
@@ -109,6 +109,24 @@
...
+        var cookies = [],
+            foundCookies = document.cookie.match(/(?:bs.table.)(\w*)/g);
+
+        if (foundCookies) {
+            $.each(foundCookies, function (i, cookie) {
+                if (/./.test(cookie)) {
+                    cookie = cookie.split(".").pop();
+                }
+
+                if (cookies.indexOf(cookie) === -1) {
```
Comment (wenzhixin) [inline src/extensions/filter-control/bootstrap-table-filter-control.js]: IE8 doesn't support this, use inArray instead.

Comment (ioctaptceb) [inline src/extensions/filter-control/bootstrap-table-filter-control.js]: yep, sorry for the delay

**Pull request merged**